### PR TITLE
Lesson 8

### PR DESCRIPTION
**Pull request by taras-kobetiak**

### CODE REVIEW

File: src/app/add-course-page/add-course-page/add-course-page.component.html [hunk 1]
```diff
@@ -21,7 +21,7 @@ <h5>Description:</h5>
                 <div class="subtitle">
                     <h5>Date:</h5>
                 </div>
-                <app-calendar [creationDate]="course.creationDate" (creationDateChange)="creationDateChange($event)">
+                <app-calendar [creationDate]="dataToSendInCalendar" (creationDateChange)="creationDateChange($event)">
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.html]: not well variable name)

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 2]
```diff
@@ -20,25 +27,23 @@ export class AddCoursePageComponent implements OnInit {
...
+      this.takeCourseData();
+    }
   }
 
   onSubmit(): void {
     this.course.id ? this.updateCourse() :
-      this.newCourse()
-
-    this.router.navigate(['courses'])
+      this.newCourse();
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: addNewCourse would be better name

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 3]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
     this.authors = authors;
   }
 
-  updateCourse(): void {
-    this.course.creationDate = new Date(this.course.creationDate);
-    this.courseService.updateCourse(this.course);
+  async updateCourse(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: you don't need to do it async especially declare return type as Promise<void> when you don't return anything)
it should looks as something like this:
```
 updateCourse(): void {
    this.courseService.updateCourse(this.course).then(() => this.router.navigate(['/courses']));
  }
```

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 4]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+    await this.courseService.updateCourse(this.course);
+    this.router.navigate(['/courses']);
   }
 
-  newCourse(): void {
-    this.course.id = this.courseService.getCoursesList().length + 1 + ''
-    this.course.creationDate = new Date(this.course.creationDate);
+  async newCourse(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: the same here, please rewrite

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 5]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+    this.courses = courseData;
+
+    this.generateId();
+    this.course.id = this.temporaryId + '';
     this.courseService.addCourses(this.course);
+    this.router.navigate(['/courses']);
+  }
+
+  generateId(): void {
+    if (!this.courses.find(course => course.id === this.temporaryId + '')) {
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: this.temporaryId.toString() or `${this.temporaryId}` but definitely not this.temporaryId + ''

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 6]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+      return;
+    } else {
+      ++this.temporaryId;
+      this.generateId();
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: recursion just for incrementation, really?)

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 7]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+    }
+  }
+
+  async takeCourseData(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: the same as updateCourse
and check all places where you used Promise<void> as return type and rewrite them

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 8]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+    let courseData = await this.courseService.getCourseById(this.courseId);
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: I believe this variable is redundant

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 9]
```diff
@@ -53,14 +58,34 @@ export class AddCoursePageComponent implements OnInit {
...
+    this.course = courseData;
+    this.course.creationDate = new Date(this.course.creationDate);
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: really strange manipulations

File: src/app/add-course-page/calendar/calendar.component.ts [hunk 10]
```diff
@@ -1,21 +1,23 @@
...
+  @Input() creationDate: Date | string;
   @Output() creationDateChange: EventEmitter<Date> = new EventEmitter()
 
   placeholderData: string;
 
-  ngOnInit(): void {
-    if (this.creationDate) {
-      Number(this.creationDate) - Number(new Date()) > -100 && Number(this.creationDate) - Number(new Date()) < 0 ? this.placeholderData = 'Choose a date' :
-        this.placeholderData = `${this.creationDate.getDate()}/${this.creationDate.getMonth()}/${this.creationDate.getFullYear()}`
+  ngDoCheck(): void {
```
Comment (artur-dorosh) [inline src/app/add-course-page/calendar/calendar.component.ts]: terrible decision, check how many times this function calls just on "edit course" button click
image

Comment (artur-dorosh) [reply src/app/add-course-page/calendar/calendar.component.ts]: 11 times
if you want wait for data and set plug data follow next way within template
image

File: src/app/app.component.ts [hunk 11]
```diff
@@ -14,8 +14,6 @@ export class AppComponent implements DoCheck {
   constructor(private authService: AuthServiceService) { }
 
   ngDoCheck(): void {
-    if (this.authService.getUserInfo().email) {
-      this.isAuth = this.authService.isAuthenticated();
-    }
+    this.isAuth = this.authService.isAuthenticated();
```
Comment (artur-dorosh) [inline src/app/app.component.ts:17]: the same here, really many calls
at least please add check
```
if (this.isAuth !== null) {
  this.isAuth = this.authService.isAuthenticated();
}
```

File: src/app/authentication/interceptor/auth.interceptor.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+
+    userData: IUserEntyty;
+    constructor(private authService: AuthServiceService) { }
+
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+
+        async () => {
+            let response = await this.authService.getUserInfo('taras@.com');
```
Comment (artur-dorosh) [inline src/app/authentication/interceptor/auth.interceptor.ts]: interceptor doesn't work with fetch but... wtf)
I just skip this file now due to future rewriting

File: src/app/authentication/services/auth-service.service.ts [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+  providedIn: 'root'
+})
+export class AuthServiceService {
+
+  usersData: IUserEntyty[]
+
+  constructor(private http: HttpClient) { }
+
+  async logIn(): Promise<any> {
+    let userData = await fetch('http://localhost:3000/users');
```
Comment (artur-dorosh) [inline src/app/authentication/services/auth-service.service.ts]: you should know what specific endpoint returns
e.g. IUserEntyty[] or something else
also you don't need async/await here
```
logIn(): Promise<IUserEntyty[]> {
  return fetch('http://localhost:3000/users').then((data) => data.json());
}
```

File: src/app/authentication/services/auth-service.service.ts [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+  logOut(): void {
+    localStorage.removeItem('currentUser');
+  }
+
+  isAuthenticated(): boolean {
+    return Boolean(localStorage.getItem('currentUser'))
+  }
+
+  async getUserInfo(email: string): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/authentication/services/auth-service.service.ts]: the same as logIn

File: src/app/breadcrumbs/breadcrumbs.component.ts [hunk 15]
```diff
@@ -25,10 +25,19 @@ export class BreadcrumbsComponent implements DoCheck {
...
-      this.breadcrumbpsTitle = ''
+      this.breadcrumbsTitle = '';
+    }
+  }
+
+  async setBreadcrumbs(): Promise<void> {
+    if (this.breadcrumbsTitle) {
+      return;
+    } else {
+      let courseData = await this.courseService.getCourseById(this.courseId);
```
Comment (artur-dorosh) [inline src/app/breadcrumbs/breadcrumbs.component.ts]: redundant variable

File: src/app/breadcrumbs/breadcrumbs.component.ts [hunk 16]
```diff
@@ -25,10 +25,19 @@ export class BreadcrumbsComponent implements DoCheck {
...
-      this.course = this.courseService.getCourseById(this.courseId) || this.defaultCourseData;
-      this.breadcrumbpsTitle = ` / ${this.course.title}`;
+      this.setBreadcrumbs();
     } else {
-      this.breadcrumbpsTitle = ''
+      this.breadcrumbsTitle = '';
+    }
+  }
+
+  async setBreadcrumbs(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/breadcrumbs/breadcrumbs.component.ts]: you don't need async/await here
please use "then" method of promises

File: src/app/header/header.component.html [hunk 17]
```diff
@@ -10,7 +10,7 @@ <h1>VIDEO COURSE</h1>
     <div class="login-form" *ngIf="isAuth">
         <div class="user-login">
             <img src="https://www.neuro-cardiolab.com.ua/img/%D0%B1%D0%B5%D0%B7%D1%84%D0%BE%D1%82%D0%BE.png" alt="">
-            <p *ngIf="isAuth; else startScreen">Hello, {{currentUserName}}</p>
+            <p *ngIf="isAuth; else startScreen">Hello, {{currentUser.firstName}}</p>
```
Comment (artur-dorosh) [inline src/app/header/header.component.html]: don't forget use "?" to avoid errors

File: src/app/header/header.component.ts [hunk 18]
```diff
@@ -9,15 +10,18 @@ import { AuthServiceService } from './services/auth-service.service';
...
-  currentUserName: string;
+  currentUser: IUserEntyty;
 
   constructor(private authService: AuthServiceService) { }
 
   ngDoCheck(): void {
-    this.currentUser = this.authService.getUserInfo().email;
     this.isAuth = this.authService.isAuthenticated();
-    this.currentUserName = this.currentUser.split('@')[0];
+    let userData: string | null = localStorage.getItem('currentUser');
```
Comment (artur-dorosh) [inline src/app/header/header.component.ts:19]: why you don't do this using ngOnInit? DoCheck calling too often

File: src/app/login-page/login-page/login-page.component.ts [hunk 19]
```diff
@@ -1,23 +1,41 @@
...
+  usersData: IUserEntyty[];
+  currentUser: IUserEntyty;
 
   constructor(private authService: AuthServiceService, private router: Router) { }
 
-  onSubmit(form: NgForm): void {
-    this.userInfo = form.value;
-    this.authService.logIn(this.userInfo);
-    this.router.navigate(['/courses'])
+  async onSubmit(form: NgForm): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/login-page/login-page/login-page.component.ts]: rewrite

File: src/app/login-page/login-page/login-page.component.ts [hunk 20]
```diff
@@ -1,23 +1,41 @@
...
+    this.currentUser = form.value;
+    this.createUsersData();
+  }
+
+  async createUsersData(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/login-page/login-page/login-page.component.ts]: rewrite

File: src/app/pages-block/pages-block.component.ts [hunk 21]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
   constructor(private coursesPagesService: CoursesService) { }
 
   ngOnInit(): void {
-    this.courses = this.coursesPagesService.getCoursesList()
+    this.refreshCourse();
+    this.allCoursesLengthFunction();
   }
 
-  deleteComponent(id: string): void {
+  async deleteComponent(id: string): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: rewrite

File: src/app/pages-block/pages-block.component.ts [hunk 22]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
   courses: ICoursePage[] = [];
+  numberOfCourses: number = 3;
+  allCoursesLength: number;
 
   constructor(private coursesPagesService: CoursesService) { }
 
   ngOnInit(): void {
-    this.courses = this.coursesPagesService.getCoursesList()
+    this.refreshCourse();
+    this.allCoursesLengthFunction();
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: try to refuse of this function

File: src/app/pages-block/pages-block.component.ts [hunk 23]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
-      this.courses = this.coursesPagesService.deleteCourse(id)
+      await this.coursesPagesService.deleteCourse(id);
+      this.allCoursesLengthFunction();
+      this.refreshCourse();
     }
   }
 
-  loadNewCourses(): void {
-    console.log('here is come action');
+  async loadNewCourses(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: rewrite

File: src/app/pages-block/pages-block.component.ts [hunk 24]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
 
   changeRate(course: ICoursePage): void {
     course.topRated = !course.topRated;
     this.coursesPagesService.updateCourse(course);
   }
 
-  findClick(inputData: string): void {
-    if (inputData !== '') {
-      this.courses = this.courses.filter(course => course.title.toLowerCase().includes(inputData.toLowerCase()))
+  async findClick(inputData: string): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: rewrite

File: src/app/pages-block/pages-block.component.ts [hunk 25]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
+    if (inputData) {
+      let findData = inputData.toLowerCase();
+      this.courses = await this.coursesPagesService.getFilteredList(findData);
     } else {
-      this.courses = this.coursesPagesService.getCoursesList();
+      this.refreshCourse();
     }
   }
+
+  async refreshCourse(): Promise<void> {
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: rewrite

File: src/app/pages-block/services/courses.service.ts [hunk 26]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
-    new CoursePage('2', 'Second', new Date(2025, 5, 30), 59, 'Not good course.'),
-    new CoursePage('3', 'Third', new Date(2022, 6, 4), 96, 'Some description   elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwersit amet consectetur, adipisicing elit. lorem lowerwer sit amet consectetur, adipisicing elit. lorem lowerwer Asperiores, est! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod, itaque.'),
-  ];
   course: CoursePage;
 
-  getCoursesList(): CoursePage[] {
-    return this.courses
+  constructor(private http: HttpClient) { }
+
+  async getCoursesList(courseLimit: number): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: rewrite as logIn

File: src/app/pages-block/services/courses.service.ts [hunk 27]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
+    let serverData = await fetch(`http://localhost:3000/courses?_limit=${courseLimit}`);
+    return await serverData.json();
+  }
+
+  async getAllCoursesList(): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: and here

File: src/app/pages-block/services/courses.service.ts [hunk 28]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
-    return this.courses = this.courses.filter(el => el.id !== id);
+  deleteCourse(id: string): Promise<any> {
+    return fetch(`http://localhost:3000/courses/${id}`, {
+      method: 'DELETE',
+    })
   }
 
-  getCourseById(id: string): CoursePage | undefined {
-    return this.courses.find(course => course.id === id)
+  async getCourseById(id: string): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: and here

File: src/app/pages-block/services/courses.service.ts [hunk 29]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
+    let serverData = await fetch(`http://localhost:3000/courses/${id}`);
+    return await serverData.json();
   }
 
-  updateCourse(course: CoursePage): void {
-    let index = this.courses.findIndex(item => item.id === course.id)
-    this.courses[index] = course;
+  updateCourse(course: CoursePage): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: don't use any

File: src/app/pages-block/services/courses.service.ts (hunk 28 above)
Comment (artur-dorosh) [reply src/app/pages-block/services/courses.service.ts]: don't use any

File: src/app/pages-block/services/courses.service.ts [hunk 30]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8'
+      },
+      body: JSON.stringify(course)
+    })
   }
 
-  deleteCourse(id: string): CoursePage[] {
-    return this.courses = this.courses.filter(el => el.id !== id);
+  deleteCourse(id: string): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: don't use any

File: src/app/pages-block/services/courses.service.ts [hunk 31]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
+  }
+
+  async getFilteredList(searchData: string): Promise<any> {
+    let serverData = await fetch(`http://localhost:3000/courses?q=${searchData}`);
+    return await serverData.json();
   }
 
-  addCourses(course: CoursePage): void {
-    this.courses = this.courses.concat(course)
+  addCourses(course: CoursePage): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: don't use any

File: src/app/pages-block/services/courses.service.ts [hunk 32]
```diff
@@ -6,31 +7,53 @@ import { CoursePage } from 'src/app/interfaces/classes';
...
+    let serverData = await fetch(`http://localhost:3000/courses?_limit=${courseLimit}`);
+    return await serverData.json();
+  }
+
+  async getAllCoursesList(): Promise<any> {
+    let serverData = await fetch(`http://localhost:3000/courses`);
+    return await serverData.json();
+  }
+
+  async getFilteredList(searchData: string): Promise<any> {
```
Comment (artur-dorosh) [inline src/app/pages-block/services/courses.service.ts]: and here
don't use any

File: src/app/pages-block/pages-block.component.ts [hunk 33]
```diff
@@ -13,33 +13,50 @@ export class PagesBlockComponent implements OnInit {
...
     course.topRated = !course.topRated;
     this.coursesPagesService.updateCourse(course);
   }
 
-  findClick(inputData: string): void {
-    if (inputData !== '') {
-      this.courses = this.courses.filter(course => course.title.toLowerCase().includes(inputData.toLowerCase()))
+  async findClick(inputData: string): Promise<void> {
+    if (inputData) {
+      let findData = inputData.toLowerCase();
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: redundant variable

**Review by artur-dorosh — CHANGES_REQUESTED** [review]

File: src/app/add-course-page/add-course-page/add-course-page.component.ts (hunk 5 above)
Comment (taras-kobetiak) [reply src/app/add-course-page/add-course-page/add-course-page.component.ts]: тут, до речі, цікавинка. ${this.temporaryId} штука не робить, воно чогось не формує строку з того

File: src/app/add-course-page/add-course-page/add-course-page.component.ts (hunk 6 above)
Comment (taras-kobetiak) [reply src/app/add-course-page/add-course-page/add-course-page.component.ts]: а чим рекурсія погана? я так старався десь її втикнути. а що,якийсь дувайл цикл краще буде?

File: src/app/add-course-page/add-course-page/add-course-page.component.ts [hunk 34]
```diff
@@ -58,34 +59,31 @@ export class AddCoursePageComponent implements OnInit {
...
+        this.router.navigate(['/courses']);
+            })
   }
 
   generateId(): void {
-    if (!this.courses.find(course => course.id === this.temporaryId + '')) {
-      return;
-    } else {
+    while(this.courses.find(course => course.id === this.temporaryId.toString())) {
       ++this.temporaryId;
```
Comment (artur-dorosh) [inline src/app/add-course-page/add-course-page/add-course-page.component.ts]: please use post incrementation

File: src/app/add-course-page/calendar/calendar.component.ts [hunk 35]
```diff
@@ -1,18 +1,18 @@
...
-export class CalendarComponent implements DoCheck {
+export class CalendarComponent implements  OnInit {
 
   @Input() creationDate: Date | string;
   @Output() creationDateChange: EventEmitter<Date> = new EventEmitter()
 
   placeholderData: string;
 
-  ngDoCheck(): void {
+  ngOnInit() {
```
Comment (artur-dorosh) [inline src/app/add-course-page/calendar/calendar.component.ts]: you don't need this code block

File: src/app/authentication/interceptor/auth.interceptor.ts [hunk 36]
```diff
@@ -11,18 +11,13 @@ export class AuthInterceptor implements HttpInterceptor {
...
-
-        async () => {
-            let response = await this.authService.getUserInfo('taras@.com');
-            let userData = await response.json();
-            this.userData = userData;
-        }
+      this.authService.getUserInfo('taras@.com')
+        .then((userData)=>  this.userData = userData[0])
         let userToken = this.userData.token;
         console.log(userToken);
```
Comment (artur-dorosh) [inline src/app/authentication/interceptor/auth.interceptor.ts]: remove

File: src/app/authentication/services/auth-service.service.ts [hunk 37]
```diff
@@ -11,21 +11,19 @@ export class AuthServiceService {
...
-  async logIn(): Promise<any> {
-    let userData = await fetch('http://localhost:3000/users');
-    return await userData.json()
+ logIn(): Promise<IUserEntyty[]> {
+return fetch('http://localhost:3000/users').then((userData: Response)=> userData.json())
   }
 
   logOut(): void {
-    localStorage.removeItem('currentUser');
+    localStorage.removeItem('currentUser')
```
Comment (artur-dorosh) [inline src/app/authentication/services/auth-service.service.ts]: why you deleted semicolons?

File: src/app/breadcrumbs/breadcrumbs.component.ts [hunk 38]
```diff
@@ -31,13 +30,14 @@ export class BreadcrumbsComponent implements DoCheck {
     }
   }
 
-  async setBreadcrumbs(): Promise<void> {
+ setBreadcrumbs(): void {
     if (this.breadcrumbsTitle) {
```
Comment (artur-dorosh) [inline src/app/breadcrumbs/breadcrumbs.component.ts]: you can simplify it
remove if statement and use else block

File: src/app/login-page/login-page/login-page.component.ts [hunk 39]
```diff
@@ -18,24 +16,25 @@ export class LoginPageComponent {
...
     if (!this.usersData.find(user => user.email === this.currentUser.email
       && user.password === this.currentUser.password)) {
       alert('wrong data, please check your email and pass');
     } else {
-      let userInfo = await this.authService.getUserInfo(this.currentUser.email);
-      this.currentUser = userInfo[0];
-
-      localStorage.setItem(`currentUser`, JSON.stringify(this.currentUser));
-      this.router.navigate(['/courses']);
+      let userInfo = this.authService.getUserInfo(this.currentUser.email)
```
Comment (artur-dorosh) [inline src/app/login-page/login-page/login-page.component.ts]: why you need this variable?

File: src/app/pages-block/pages-block.component.ts [hunk 40]
```diff
@@ -20,43 +22,43 @@ export class PagesBlockComponent implements OnInit {
...
+        this.refreshCourse();
+      })
     }
   }
 
-  async loadNewCourses(): Promise<void> {
-    this.numberOfCourses += 3;
-    await this.refreshCourse();
+loadNewCourses(): void {
+  this.numberOfCourses += 3;
```
Comment (artur-dorosh) [inline src/app/pages-block/pages-block.component.ts]: magic number antipattern
move value to constant

**Review by artur-dorosh — CHANGES_REQUESTED** [review]

**Review by artur-dorosh — APPROVED** [review]

**Pull request merged**